lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -454,6 +524,64 @@ downcastTxOut (TxOutInRecentEra addr val Alonzo.NoDatum Nothing)
 downcastTxOut (TxOutInRecentEra addr val (Alonzo.DatumHash dh) Nothing)
     = Right $ Alonzo.TxOut addr val (SJust dh)
 
+--
+-- MinimumUTxO
+--
+
+-- |
+--
+-- prop> forall out.
+-- prop> let
+-- prop>     c = computeMinimumCoinForUTxO out
+-- prop> in
+-- prop>     forall c' > c. satisfiesMinimumUTxO (out { coin = c' })
+--
+-- NOTE:
+computeMinimumCoinForTxOut
+    :: forall era. RecentEra era -- FIXME [ADP-2353] Replace 'RecentEra' with 'IsRecentEra'
+    -> Core.PParams (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+    -> Coin
+computeMinimumCoinForTxOut era pp out = withCLIConstraint era $
+    evaluateMinLovelaceOutput pp (withMaxLengthSerializedCoin out)
+  where
+    withMaxLengthSerializedCoin
+        :: TxOut (ShelleyLedgerEra era)
+        -> TxOut (ShelleyLedgerEra era)
+    withMaxLengthSerializedCoin =
+        withStandardCryptoConstraint era $
+            modifyTxOutValue era $ modifyCoin (const $ toLedger txOutMaxCoin)
+
+isBelowMinimumCoinForTxOut
+    :: forall era. RecentEra era -- FIXME [ADP-2353] Replace 'RecentEra' with 'IsRecentEra'
+    -> Core.PParams (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+    -> Either Coin ()
+isBelowMinimumCoinForTxOut era pp out = withCLIConstraint era $
+    let
+        requiredMin = evaluateMinLovelaceOutput pp out
+        actualCoin = getCoin era out
+    in
+        if actualCoin >= requiredMin
+        then Right ()
+        else Left $ requiredMin <-> actualCoin
+  where
+    getCoin :: RecentEra era -> TxOut (ShelleyLedgerEra era) -> Coin
+    getCoin RecentEraBabbage (Babbage.TxOut _ val _ _) = coin val
+    getCoin RecentEraAlonzo (Alonzo.TxOut _ val _) = coin val
+
+setMinimumCoinForTxOut
+    :: forall era. RecentEra era -- FIXME [ADP-2353] Replace 'RecentEra' with 'IsRecentEra'
+    -> Core.PParams (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+setMinimumCoinForTxOut era pp out =
+    let
+        minCoin = computeMinimumCoinForTxOut era pp out
+    in
+        withStandardCryptoConstraint era $
+            modifyTxOutValue era (modifyCoin (const minCoin)) out
```

Comment:
You can save 2 lines and 1 level of indentation here:

```suggestion
setMinimumCoinForTxOut era pp out = withStandardCryptoConstraint era $
    modifyTxOutValue era (modifyCoin (const minCoin)) out
  where
    minCoin = computeMinimumCoinForTxOut era pp out
```

Reply:
I think this is a case where I do prefer sacrificing lines and indentation 😁. My thinking is that the definition of `minCoin` is important to understanding the rest of the implementation of `setMinimumCoinForTxOut`, so it should be read _before_ reading the next line(s).

Reply:
Putting the `withStandardCryptoConstraint ` on the first line might be an idea though...